tests/addons/test_addon.py

```diff
@@ -97,6 +98,27 @@ def test_additional_catalogue_src_name_validation():
         addon._validate_additional_catalogue_srcs(metadata)
 
 
+def test_secret_names_validation():
+    addon = Addon(addon_with_secrets_path(), "stage")
+    metadata = addon.metadata
+    duplicate = metadata["secrets"][0]
+    metadata["secrets"].append(duplicate)
+    with pytest.raises(AddonLoadError):
+        addon._validate_secret_names(metadata)
```

Comment:
After the above-suggested change in the method:

```suggestion
    addon = Addon(addon_with_secrets_path(), "stage")
    duplicate = metadata["secrets"][0]
    addon.metadata["secrets"].append(duplicate)
    with pytest.raises(AddonLoadError):
        addon._validate_secret_names()
```